### ISSUE TITLE
deployment_token provided was invalid. on 2 workflow files for same repo.

### ISSUE THREAD

**Issue by jose-vazquez2** [Azure/static-web-apps#718]
**Describe the bug**

Have 2 environments mapped to 2 branches prod and dev, i deployed 1 static web app for prod and that worked great, but when i added a new workflow file for dev linked to an entire new static web app wit its own workflow  file and secret locked to develop branch the build and deploy step fails with "deployment_token provided was invalid."  i did double check the deployment_token was correct for each workflow file, ( i know about the staging env feature but that doesnt work well with our auth so we going the 2 static web apps route)

**To Reproduce**
create 2 static web apps, linked to the same repository but different branch.
Push to main branch deploys goes fine.
Push to develop branch  deploy fails with "deployment_token provided was invalid"

the workflow file is the auto-generated for github actions on https://docs.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions
with the branch modified 

```
on:
  push:
    branches:
      - develop
  pull_request:
    types: [opened, synchronize, reopened, closed]
    branches:
      - develop

```

**Expected behavior**
Each branch should deploy to each static web app without any issues.

**Comment by miwebst** [comment]
Hi @jose-vazquez2, we will need more information to investigate this. If you are comfortable, can you post your complete workflow files, the default hostname of your 2 SWA's, and any relevant logs from the GitHub Actions output? If you are not comfortable posting that, I'd recommend going through support in the Azure Portal.

Another thing you can do is reset the deployment token in the Azure Portal for the SWA that is not working and update the secret in your GitHub repo. You should not see this from a fresh deployment token.

**Comment by jose-vazquez2** [comment]
Hi @miwebst, i did try refreshing the deployment token multiple times on both apps, still same behaviour, also tried creating an entire new static web app for dev and got same result

the workflow file is exactly the same as the one auto-generated by their wizard in both just swap the branch name and the secret.key  for each env (i renamed the secret key name for convenience, but the exact same behavior was with the autogenerated AZURE_STATIC_WEB_APPS_API_TOKEN_SOME_RANOM_ID)

```
name: Azure Static Web Apps CI/CD

on:
  push:
    branches:
      - develop
  pull_request:
    types: [opened, synchronize, reopened, closed]
    branches:
      - develop

jobs:
  build_and_deploy_job:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: self-hosted
    name: Build and Deploy Job
    steps:
      - uses: actions/checkout@v2
        with:
          submodules: true
      - name: Build And Deploy
        id: builddeploy
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_DEV }}
          repo_token: ${{ secrets.GITHUB_TOKEN }} # Used for Github integrations (i.e. PR comments)
          action: "upload"
          ###### Repository/Build Configurations - These values can be configured to match your app requirements. ######
          # For more information regarding Static Web App workflow configurations, please visit: https://aka.ms/swaworkflowconfig
          app_location: "/" # App source code path
          api_location: "" # Api source code path - optional
          output_location: "" # Built app content directory - optional
          ###### End of Repository/Build Configurations ######

  close_pull_request_job:
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
    runs-on: self-hosted
    name: Close Pull Request Job
    steps:
      - name: Close Pull Request
        id: closepullrequest
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_DEV}}
          action: "close"
```

**Comment by miwebst** [comment]
I'm not able to reproduce this. I just created this repository, made a second branch called 'develop' then created an SWA for each branch: https://github.com/miwebst/testdeploy 

I'd suggest trying to create a SWA against that branch directly through the Portal or VS Code so it autogenerates the workflow file. If that is messed up, something larger is at play. If that works, then it seems to be a configuration issue.

**Comment by jose-vazquez2** [comment]
it happens using both initially i was running everything using vscode after that didnt work and i spend like  day trying to figure out what was going on, i switched to make it from the web portal and also didnt work.

btw here is a piece of the log where it fails, i did verify the token being used is the correct one by printing it in to the same log file, but i omited that part here.

```
Run Azure/static-web-apps-deploy@v1
  with:
    azure_static_web_apps_api_token: ***
    repo_token: ***
    action: upload
    app_location: /

DeploymentId: ec9b8902-5ee6-4d8c-b9d7-7b217a4fc519

deployment_token provided was invalid.
An unknown exception has occurred

For further information, please visit the Azure Static Web Apps documentation at https://docs.microsoft.com/en-us/azure/static-web-apps/
If you believe this behavior is unexpected, please raise a GitHub issue at https://github.com/azure/static-web-apps/issues/
Exiting

```

they are running on a dockerized self-hosted container that does the deployment

**Comment by miwebst** [comment]
Oh so this is not running on GitHub Actions but on a self-hosted container? Can you try running it on the regular GitHub infrastructure and see if that fixes it? You could also try clearing all the docker containers in case it has cached an old container.

**Comment by jose-vazquez2** [comment]
hmm unfortunately i cannot use github cloud runners, company policy, but i will look into the self-hosted, you think the main issue lies there? 
will look into it and report back what i find.

**Comment by jose-vazquez2** [comment]
i spent a few days digging into the container, to see if i found something odd, but the container seems good.
i suspect the deployment token for azure might not have necessary permissions.
does anyone know if is there a way to check the permissions or validate the deployment token using CLI command?
az getInfo <deployment_token>

at this point im lost at to what could be failing.

**Comment by jose-vazquez2** [comment]
i was able to make it work,  could not find the source of the issue, but what i did was to Destroy the self-hosted runner and re-build it from scratch, after that i was able to run the workflows fine.

**Comment by Richard-Marlow** [comment]
I have this exact same issue that I have been trying to figure out for days. "deployment_token provided was invalid." when trying to deploy to our dev environment using our self hosted dev runner. When I deploy to UAT (which uses a different self hosted runner) it works without any issues.

After reading through this thread I switched from a self hosted runner to a github hosted runner. The job worked perfectly, without error. 

I then unregistered our self hosted runner and uninstalled it. There was not a trace of it left on the VM we hosted it on. I ran through the install process again, creating a new runner with a new name. After switching the workflow back to use the new self hosted runner, it failed again with "deployment_token provided was invalid."
